src/dxtbx/format/FormatPilatusHelpers.py

```diff
@@ -3,29 +3,25 @@
 identifying the regions to be masked.
 """
 
+from dataclasses import dataclass
+from typing import List, Tuple
 
-import collections
-
-_Detector = collections.namedtuple(
-    "Detector",
-    [
-        "module_size_fast",
-        "module_size_slow",
-        "gap_fast",
-        "gap_slow",
-        "module_configs_fast",
-    ],
-)
-setattr(
-    _Detector,
-    "all_widths",
-    property(
-        lambda self: [
+
+@dataclass
```

Comment:
Since this is replacing a tuple, shouldn't it be
```suggestion
@dataclass(frozen=True)
```
so it's clear they shouldn't be altered at runtime? (or `typing.NamedTuple`, but the nuance of that vs frozen-dataclass is a choice I don't understand in this context yet)

Reply:
I think they _are_ altered at runtime after construction, e.g. here:
https://github.com/cctbx/dxtbx/blob/923f92d167087029fc04b44460304aba7983ac74/src/dxtbx/format/FormatPilatusHelpers.py#L200-L205

Reply:
Let's try it

Reply:
Sorry, that was me be dumb, I saw ` = ` instead of ` == `. I think you're correct - the previous usage of `collections.namedtuple` should have prevented modification of the attributes before, so setting `@dataclass(frozen=True)` should be equivalent behaviour.

Reply:
👍